app/ts/utils/bigint.ts

```diff
@@ -139,3 +139,11 @@ export function calculateWeightedPercentile(data: readonly { dataPoint: bigint,
 	const interpolation = (targetIndex - lowerWeight) / (upperWeight - lowerWeight)
 	return lowerValue.dataPoint + (upperValue.dataPoint - lowerValue.dataPoint) * interpolation
 }
+
+export const bigintSecondsToDate = (seconds: bigint) => {
+	if (seconds > Number.MAX_SAFE_INTEGER) throw new Error(`Too big seconds value: ${ seconds }`)
```

Comment:
Suggest removing this restriction for a few reasons:
1. A very big number will just get rounded, and I don't think anyone will care that a date millions of years in the future is rounded by a second or two.
2. `new Date(Number.MAX_SAFE_INTEGER * 1000)` results in an error because the Date object can't handle dates that far in the future.
3. You are multiplying this by 1000 below, so this doesn't actually protect against rounding.
You could address each of these, but it doesn't seem worth it.  If you do want to address them, then I recommend figuring out what the max safe date seconds is and using that as a restriction instead.  https://stackoverflow.com/a/27093173 suggests max safe date **seconds** is `8,640,000,000,000` which is *way* below `MAX_SAFE_INTEGER`.

```suggestion
```
```suggestion
	if (seconds > 8640000000000n) throw new Error(`Too big seconds value: ${ seconds }`)
```

Reply:
it protects cases where the seconds are just too big for some probably accidental reason, but yeah, we can make it more strict